src/PolyfillTrait6.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+/*
+ * This file is part of PHPUnit Good Practices.
+ *
+ * (c) Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PHPUnitGoodPractices\Polyfill;
+
+/**
+ * @internal
+ */
+trait PolyfillTrait6
+{
+    public function expectException($exception)
+    {
+        if (\is_callable(['parent', 'expectException'])) {
+            parent::expectException($exception);
+        } else {
+            $this->setExpectedException($exception);
+        }
+    }
+
+    public function expectExceptionMessageMatches($regexp)
+    {
+        if (\is_callable(['parent', 'expectExceptionMessageMatches'])) {
+            parent::expectExceptionMessageMatches($regexp);
+        } elseif (\is_callable(['parent', 'setExpectedExceptionRegExp'])) {
+            $this->setExpectedExceptionRegExp(\Exception::class, $regexp);
+        } else {
+            $this->expectExceptionMessageRegExp($regexp);
+        }
+    }
+
+    public static function assertIsArray($actual, $message = '')
+    {
+        static::assertInternalType('array', $actual, $message);
```

Comment:
shouldn't we call `parent::assertIsArray` if it exists ?

Reply:
Is there an actual need to do that? Every version that doesn't have `assertIsArray` has `assertInternalType`, and, likewise, every version that does have them both, knows how to juggle the calls without potentially expensive `is_callable` checks.

What I mean is that PHPUnit knows better when to call  `assertIsArray` instead of `assertInternalType` when it has them both. It just doesn't make sense to do PHPUnit's job here.

Reply:
I believe that there IS a need.

```php
<?php

class BaseClass {
    public function exampleMethod_A() {
        var_dump(__METHOD__);
    }
    public function exampleMethod_B() {
        var_dump(__METHOD__);
    }
}

trait PolyfillTrait {
  // we are OVERRIDING the method
  public function exampleMethod_A() {
      var_dump(__METHOD__);
  }

  // we are POLYFILLING the method
  public function exampleMethod_B() {
      if (\is_callable(["parent", __FUNCTION__])) {
          parent::exampleMethod_B();
      } else {
          var_dump(__METHOD__);
      }
  }
}

class MyClass extends BaseClass {
    use PolyfillTrait;
}

(new MyClass())->exampleMethod_A(); // result: "PolyfillTrait::exampleMethod_A" << wrong! we should use original implementation!
(new MyClass())->exampleMethod_B(); // result: "BaseClass::exampleMethod_B" << OK, we called existing method
```

our package should polyfill what is missing, not override everything

Reply:
Fine, you are the maintainer here. If you want the trouble dealing with endless `else` that exists for no practical reason, that can't be tested, so be it. Done.

Reply:
there is a reason. main implementation can be different than the one from override. or it can be already patched by something else, and here you would bypass that patch.

Reply:
I stand corrected. You're right. If we consider all possible use-cases, then, indeed, calling the parent is the right choice.